tests/Schema/Provider/FromFilesSchemaProvider/FromFilesSchemaProviderTest.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Cycle\Tests\Schema\Provider\FromFilesSchemaProvider;
+
+use InvalidArgumentException;
+use LogicException;
+use PHPUnit\Framework\TestCase;
+use Yiisoft\Aliases\Aliases;
+use Yiisoft\Yii\Cycle\Schema\Provider\FromFilesSchemaProvider;
+
+class FromFilesSchemaProviderTest extends TestCase
+{
+    public function getWithConfigEmptyData(): array
+    {
+        return [
+            [
+                [],
+            ],
+            [
+                ['files' => []],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider getWithConfigEmptyData
+     *
+     * @param array $config
+     */
+    public function testWithConfigEmpty(array $config): void
+    {
+        $schemaProvider = $this->createSchemaProvider();
+
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Files not set.');
+        $schemaProvider->withConfig($config);
+    }
+
+    public function testWithConfigInvalidData(): void
+    {
+        $schemaProvider = $this->createSchemaProvider();
+
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('The "files" parameter must be an array.');
+        $schemaProvider->withConfig(['files' => '@dir/schema1.php']);
+    }
+
+    public function testWithConfig(): void
+    {
+        $schemaProvider = $this->createSchemaProvider();
+
+        $data = $schemaProvider
+            ->withConfig(['files' => ['@dir/schema1.php']])
+            ->read();
+
+        $this->assertSame([
+            'user' => [],
+        ], $data);
+    }
+
+    public function testWithConfigImmutable(): void
+    {
+        $schemaProvider1 = $this->createSchemaProvider();
+        $schemaProvider2 = $schemaProvider1->withConfig([
+            'files' => ['@dir/schema1.php'],
+        ]);
+        $this->assertNull($schemaProvider1->read());
+        $this->assertSame([
+            'user' => [],
+        ], $schemaProvider2->read());
+    }
+
+    public function testRead(): void
+    {
+        $schemaProvider = $this->createSchemaProvider();
+
+        $data = $schemaProvider
+            ->withConfig([
+                'files' => [
+                    '@dir/schema1.php',
+                    '@dir/schema-not-exists.php', // not exists files should be silent
```

Comment:
Why not throw an exception if the file is missing?

Reply:
According to the logic described [here](https://github.com/yiisoft/yii-cycle/blob/master/docs/en/reading-schema.md) there should be no exception.

Reply:
Add `strict` parameter.